src/Streamly/Internal/Data/Fold.hs

```diff
@@ -1227,7 +1212,38 @@ span
     -> Fold m a b
     -> Fold m a c
     -> Fold m a (b, c)
-span p fld1 fld2 = splitWith (,) (ltakeWhile p fld1) fld2
+span p (Fold stepL initialL extractL) (Fold stepR initialR extractR) =
```

Comment:
Make this a parser operation and use splitWith, takeWhile parser ops to implement.

Reply:
Let's leave this for now, let's not remove things that are working properly. We can always remove stuff later.